src/dmd/expression.d

```diff
@@ -4668,12 +4668,14 @@ extern (C++) final class DotVarExp : UnaExp
 
     override bool isLvalue()
     {
-        return true;
+        return e1.op != TOK.structLiteral;
```

Comment:
The problem with structLiteral is not that wether they are Lvalue or not it's rather that when they are it's not mutable.